### PR TITLE
add job peek from storage

### PR DESCRIPTION
**Pull request by Napoleon0621**
#169

### CODE REVIEW

File: storage/persistence/spanner/spanner.go [hunk 1]
```diff
@@ -163,6 +163,35 @@ func (s *Spanner) GetReadyJobs(ctx context.Context, req *model.JobDataReq) (jobs
...
 
+// BatchGetJobsByID returns job data by job ID
+func (s *Spanner) BatchGetJobsByID(ctx context.Context, IDs []string) (jobs []*model.JobData, err error) {
+	txn := s.cli.ReadOnlyTransaction()
+	defer txn.Close()
+
+	for _, id := range IDs {
+		iter := txn.Query(ctx, spanner.Statement{
+			SQL: "SELECT pool_name, job_id, namespace, queue, body, ready_time, expired_time, created_time, tries " +
+				"FROM lmstfy_jobs WHERE job_id=@jobid LIMIT @limit",
```
Comment (git-hulk) [inline storage/persistence/spanner/spanner.go:174]:
```suggestion
				"FROM lmstfy_jobs WHERE job_id in IN UNNEST(@ids)",
```

Can use `IN` condition to query jobs at once

File: storage/persistence/spanner/spanner_test.go [hunk 2]
```diff
@@ -100,3 +100,17 @@ func TestSpannerDataMgr_GetReadyJobs(t *testing.T) {
 	assert.EqualValues(t, 2, len(jobs))
 	mgr.DelJobs(ctx, jobIDs)
 }
+
+func TestSpannerDataMgr_BatchGetJobsByID(t *testing.T) {
```
Comment (git-hulk) [inline storage/persistence/spanner/spanner_test.go:104]:
```suggestion
func TestSpanner_BatchGetJobsByID(t *testing.T) {
```

**Comment by Napoleon0621** [comment]
@git-hulk Changes are made related to the above comments, please take another look, thx.

File: engine/redis_v2/engine.go [hunk 3]
```diff
@@ -243,6 +243,18 @@ func (e *Engine) Peek(namespace, queue, optionalJobID string) (job engine.Job, e
 		// return jobID with nil body if the job is expired
 		return engine.NewJobWithID(namespace, queue, nil, 0, 0, jobID), nil
 	}
+
+	// look up job data in storage
+	if err == engine.ErrNotFound && e.cfg.EnableSecondaryStorage {
+		res, err := storage.Get().GetJobByID(context.TODO(), optionalJobID)
+		if err != nil || len(res) <= 0 {
+			return nil, engine.ErrNotFound
+		}
```
Comment (git-hulk) [inline engine/redis_v2/engine.go:252]:
```suggestion
		if err != nil {
		    return nil, err
		}
		if len(res) == 0 {
			return nil, engine.ErrNotFound
		}
```

Comment (Napoleon0621) [reply engine/redis_v2/engine.go]: @git-hulk fixed, please have another look, thx.

**Review by git-hulk — APPROVED** [review]

**Pull request merged**